cmd/goal/clerk.go

```diff
@@ -123,6 +125,7 @@ func init() {
 
 	compileCmd.Flags().BoolVarP(&disassemble, "disassemble", "D", false, "disassemble a compiled program")
 	compileCmd.Flags().BoolVarP(&noProgramOutput, "no-out", "n", false, "don't write contract program binary")
+	compileCmd.Flags().BoolVarP(&writeSourceMap, "map", "m", false, "write out assembly map")
```

Comment:
@algochoi I'm asking without knowing the standards/conventions - Is it typical to write a test exercising `goal` changes?

* I'm not sure what flavors of test are considered reasonable here.  I'm loosely aware `e2e_sub`s exercise goal commands.
* If a test is to be added, I feel the intent is to confirm the command exists and produces an output matching the expected shape (e.g. contains expected keys [like `mapping`]).

Reply:
yes we write goal expect tests to check goal command line options

Reply:
@algorandskiy Gotcha - I think I found the related test area:  https://github.com/algorand/go-algorand/tree/master/test/e2e-go/cli/goal/expect.  I'll take a look to familiarize myself.

Reply:
The PR adds an assertion to an existing `e2e_sub`.  Resolving under the assumption that the change is sufficient.  If there's more discussion, let's reopen the thread.